common/dal/sample_da.py

```diff
@@ -1252,4 +1252,31 @@ def process_stale_sending_dtol_samples(self,refresh_threshold=3600):
         update_data = {"status": "processing", "update_type": "system", "updated_by": "system", 'date_modified': datetime.now(timezone.utc).replace(microsecond=0), 'time_updated': datetime.now(timezone.utc).replace(microsecond=0)}
         self.get_collection_handle().update_many(
             {"sample_type": {"$in": TOL_PROFILE_TYPES},
-                "status": "sending", "date_modified": {"$lt": datetime.now(timezone.utc).replace(microsecond=0) - timedelta(seconds=refresh_threshold)}},{"$set": update_data})
\ No newline at end of file
+                "status": "sending", "date_modified": {"$lt": datetime.now(timezone.utc).replace(microsecond=0) - timedelta(seconds=refresh_threshold)}},{"$set": update_data})
+        
+    def get_all_sample_fields(self, has_copo_defined_fields=True, has_project_defined_fields=True):
```

Comment:
The usage is to retrieve both TOL manifest defined fields which are primarily uppercase fields or fields that are uppercase with underscores.

That's true, the latest manifest version should be considered so that the latest fields are retrieved.

That's also true, we would not but I would like to retrieve them hence the creation of this function so that I can use it in the 'generate_standards_map' logic

"All sample fields" means all fields - both COPO defined which are the lowercase or camelcase fields and the project (i.e. ERGA, ASG, DTOL) defined ones i.e. the uppercase ones and uppercase with underscores.

Reply:
The sample collection also contains sample data with the type other than ERGA, ASG & DTOL.  Those non copo defined column names are also non uppercase.  We cannot generalize such rule to all samples.    This function cannot apply to all samples unless you consider the specification in the schema.  For all samples fields with project defined fields, it mixes up all the fields from different sample type, does it have its usage?